File: 파이썬/프로그래머스 Lv1/구현 - 문자열 나누기.py
def solution(s):
    answer = 0

    count_x = 0
    count_y = 0
    char_x = s[0]
    for c in s:
        if (count_x == 0):
            char_x = c

        if (char_x == c):
            count_x += 1
        else:
            count_y += 1

        if (count_x == count_y):
            answer += 1
            count_x = 0
            count_y = 0

    if (count_x != 0 or count_y != 0):
        answer += 1

    return answer


print(solution("banana"))       # 3
print(solution("abracadabra"))  # 6

File: 자바/프로그래머스LV1/구현_문자열나누기.java
 package 프로그래머스LV1;

public class 구현_문자열나누기 {

  public int solution(String s) {
    int answer = 0;

    int count_x = 0;
    int count_y = 0;

    char char_x = s.charAt(0);

    for (char c : s.toCharArray()) {
      if (count_x == 0) {
        char_x = c;
      }

      if (char_x == c) {
        count_x++;
      } else {
        count_y++;
      }

      if (count_x == count_y) {
        count_x = 0;
        count_y = 0;
        answer++;
      }
    }

    return answer;
  }

  public static void main(String[] args) {
    구현_문자열나누기 test = new 구현_문자열나누기();
    System.out.println(test.solution("banana")); //	3
    System.out.println(test.solution("abracadabra")); //	6
    System.out.println(test.solution("aaabbaccccabba")); //	3
  }

}
